javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -945,6 +946,34 @@ object Formatter {
   private val FormatSpecifier = new js.RegExp(
       """(?:(\d+)\$)?([-#+ 0,\(<]*)(\d+)?(?:\.(\d+))?[%A-Za-z]""", "g")
 
+  private final val StrsOfZerosMax = 20
+  private final val StrsOfZerosLength = 1 + StrsOfZerosMax
+
+  private val StrsOfZeros: Array[String] = {
+    val strsOfZeros = new Array[String](StrsOfZerosLength)
+    strsOfZeros(0) = ""
+    var i = 1
+    while (i != StrsOfZerosLength) {
+      strsOfZeros(i) = strsOfZeros(i - 1) + "0"
+      i += 1
+    }
+    strsOfZeros
+  }
+
+  private def strOfZeros(count: Int): String = {
+    if (count < StrsOfZerosLength) {
+      StrsOfZeros(count)
+    } else {
+      var result = ""
+      var remaining = count
+      while (count > StrsOfZerosMax) {
+        result += StrsOfZeros(StrsOfZerosMax)
+        remaining -= StrsOfZerosMax
+      }
+      result + StrsOfZeros(remaining)
+    }
+  }
+
```

Comment:
We can't have a single constant string here, because there is no reasonable bound on the number of zeros that we need. Consider for example `String.format("%.1000f", 1.0)`, which would need 1000 zeros.

Reply:
Ah sorry, I meant the `StrsOfZeroes` more specifically. Wouldn't it be sufficient to create a single zero string with `StrsOfZerosMax` zeroes? (we'll still need `strOfZeroes` of course).

Reply:
Yes, we could do that as well, indeed.